apps/scratch/src/hooks/useEmbeddedScratch.ts

```diff
@@ -222,6 +231,22 @@ export class EmbeddedScratchCallbacks {
     });
   }
 
+  async exportTask(request: ExportTask["request"]): Promise<ExportTaskResult> {
+    try {
+      const task = await getTask(this.getTask.bind(this), request.id);
```

Comment:
I am sorry for not being clear here ^^

What I think would be good is to have a function `getTaskXYZ` or whatever that is being called here AND in `async getTask(request: ...)` where it just returns

This way we have the implementation in one place and there is no need to pass requests or responses nor a need to mock them. And I checked in more detail and as far as I can see, this actually already exists and is called `saveCrtProject` :)